docs/When_Do_You_Need_A_Custom_Device.md

```diff
@@ -0,0 +1,73 @@
+Several hardware vendors have implemented custom devices for their hardware. You should check with the manufacturer that a custom device doesn’t exist before you build one.
+ 
+In general, there are three specifications that are best-suited for a custom device.
+1.	3rd Party Hardware
+2.	Unsupported Measurement or Generation Mode
+3.	Feature
+
+#### 3rd Party Hardware
+
+A list of hardware natively supported by VeriStand is found in **[VeriStand Help](https://www.ni.com/documentation/en/veristand/latest/manual/manual-overview/)** » **[VeriStand Reference](https://www.ni.com/documentation/en/veristand/latest/manual/veristand-reference/)** » **[NI Hardware Support](https://www.ni.com/documentation/en/veristand/latest/manual/ni-hardware-support/)**. If the application requires other hardware, it can probably be implemented in a custom device.
+
+#### Unsupported Measurement or Generation Mode
+
+Check **[VeriStand Help](https://www.ni.com/documentation/en/veristand/latest/manual/manual-overview/)** » **[Configuring and Running a Project](https://www.ni.com/documentation/en/veristand/latest/manual/configure-run-project/)** » **[Configuring a System Definition File](https://www.ni.com/documentation/en/veristand/latest/manual/configure-system-definition-file/)** » **[Adding and Configuring Hardware Devices](https://www.ni.com/documentation/en/veristand/latest/manual/add-configure-hardware-device/)** to determine if the required measurement or generation mode of your hardware is supported. If not, it can probably be implemented in a custom device. For example, single-point hardware-timed analog acquisition on NI-DAQ devices is supported out-of-the-box. Continuous analog acquisition can be implemented as a custom device.
+
+#### Feature
+
+All of the common functionality necessary for most real-time testing applications such as host interface communication, data logging, stimulus generation, etc, is provided by VeriStand – ready to configure and use. You should first try to meet specifications with the built-in functionality because it is engineered, tested, and supported by National Instruments.
+
+If a required feature is not built-in, it can be implemented by extending VeriStand. See NI Developer Zone Tutorial: **[Using NI VeriStand with Other Software Environments to Create Real-Time Test Applications](https://www.ni.com/en-us/innovations/white-papers/09/using-ni-veristand-with-other-software-environments-to-create-re.html)** for a complete list of ways to customize and extend VeriStand. Certain features are best implemented as custom devices. To determine when a custom device is the most appropriate mechanism to meet a specification, you should be familiar with all the customization methods available. A general rule-of-thumb is that custom devices implement features that require or use VeriStand channel data on the execution host.
```

Comment:
```suggestion
If a required feature is not built-in, it can be implemented by extending VeriStand. See **[Using NI VeriStand with Other Software Environments to Create Real-Time Test Applications](https://www.ni.com/en-us/innovations/white-papers/09/using-ni-veristand-with-other-software-environments-to-create-re.html)** for a complete list of ways to customize and extend VeriStand. Certain features are best implemented as custom devices. To determine when a custom device is the most appropriate mechanism to meet a specification, you should be familiar with all the customization methods available. A general rule-of-thumb is that custom devices implement features that require or use VeriStand channel data on the execution host.
```